test/go/src/common/clientserver_test.go

```diff
@@ -34,18 +35,17 @@ import (
 
 type test_unit struct {
 	host          string
-	port          int64
```

Comment:
this is an improvement on the tests to always use port 0 and let listener to pick an available port for us (except for http context timeout test)